autonomy/deploy/base.py

```diff
@@ -177,48 +267,49 @@ def generate_agent(
         if override_idx is None:
             override_idx = agent_n
 
-        overrides = self.process_model_args_overrides(override_idx)
-        agent_vars.update(overrides)
-        for var_name, value in agent_vars.items():
-            if any([isinstance(value, list), isinstance(value, dict)]):
-                agent_vars[var_name] = json.dumps(value)
```

Comment:
No longer required since this is automatically handles by `generate_env_vars_recursively`